packages/transaction-controller/src/utils/resimulate.ts

```diff
@@ -0,0 +1,288 @@
+    originalTransactionMeta;
+
+  const { id: transactionId, securityAlertResponse: newSecurityAlertResponse } =
+    newTransactionMeta;
+
+  if (isEqual(originalSecurityAlertResponse, newSecurityAlertResponse)) {
+    return false;
+  }
+
+  log('Security alert updated', {
+    transactionId,
+    originalSecurityAlertResponse,
+    newSecurityAlertResponse,
+  });
+
+  return (
+    newSecurityAlertResponse?.result_type === BLOCKAID_RESULT_TYPE_MALICIOUS
+  );
+}
+
+/**
+ * Determine if a transaction has a value and simulation native balance mismatch.
+ * @param originalTransactionMeta - The original transaction metadata.
+ * @param newTransactionMeta - The new transaction metadata.
+ * @returns Whether the transaction has a value and simulation native balance mismatch.
+ */
+function hasValueAndNativeBalanceMismatch(
+  originalTransactionMeta: TransactionMeta,
+  newTransactionMeta: TransactionMeta,
+): boolean {
+  const { simulationData: originalSimulationData } = originalTransactionMeta;
+
+  const { simulationData: newSimulationData, txParams: newTxParams } =
+    newTransactionMeta;
+
+  if (
+    !newSimulationData ||
+    isEqual(originalSimulationData, newSimulationData)
+  ) {
+    return false;
+  }
+
+  const newValue = newTxParams?.value ?? '0x0';
+
+  const newNativeBalanceDifference =
+    newSimulationData?.nativeBalanceChange?.difference ?? '0x0';
+
+  return !percentageChangeWithinThreshold(
+    newValue as Hex,
+    newNativeBalanceDifference,
+    false,
+    newSimulationData?.nativeBalanceChange?.isDecrease === false,
+  );
+}
+
+/**
+ * Determine if a balance change has been updated.
+ * @param originalBalanceChange - The original balance change.
+ * @param newBalanceChange - The new balance change.
+ * @returns Whether the balance change has been updated.
+ */
+function isBalanceChangeUpdated(
+  originalBalanceChange?: SimulationBalanceChange,
+  newBalanceChange?: SimulationBalanceChange,
+): boolean {
+  return !percentageChangeWithinThreshold(
+    originalBalanceChange?.difference ?? '0x0',
+    newBalanceChange?.difference ?? '0x0',
+    originalBalanceChange?.isDecrease === false,
+    newBalanceChange?.isDecrease === false,
+  );
+}
+
+/**
+ * Determine if the percentage change between two values is within a threshold.
+ * @param originalValue - The original value.
+ * @param newValue - The new value.
+ * @param originalNegative - Whether the original value is negative.
+ * @param newNegative - Whether the new value is negative.
+ * @returns Whether the percentage change between the two values is within a threshold.
+ */
+function percentageChangeWithinThreshold(
+  originalValue: Hex,
+  newValue: Hex,
+  originalNegative?: boolean,
+  newNegative?: boolean,
+): boolean {
+  let originalValueBN = new BN(remove0x(originalValue), 'hex');
+  let newValueBN = new BN(remove0x(newValue), 'hex');
+
+  if (originalNegative) {
+    originalValueBN = originalValueBN.neg();
+  }
```

Comment:
Q: (and sorry for the bother, I'm struggling to wrap my head around this) 

currently, if the balance change isDecrease === false, then the original BN is negative and we apply negative. 

are we sure this is what we want? rather than isDecrease === true, then apply the negative BN?